azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/implementation/NicIpConfigurationImpl.java

```diff
@@ -237,16 +239,22 @@ public NicIpConfigurationImpl withSubnet(String name) {
     }
 
     @Override
-    public NicIpConfigurationImpl withExistingLoadBalancer(LoadBalancer loadBalancer) {
-        this.loadBalancerToAssociate = loadBalancer;
-        return this;
+    public NicIpConfigurationImpl withExistingLoadBalancerBackend(LoadBalancer loadBalancer, String backendName) {
+        for (BackendAddressPoolInner pool : loadBalancer.inner().backendAddressPools()) {
+            if (pool.name().equalsIgnoreCase(backendName)) {
+                ensureBackendAddressPools().add(pool);
+                return this;
+            }
+        }
+
+        return null;
```

Comment:
the reason I figured returning null would be ok in scenarios involving the passing of a child's name as an argument is because it seems equivalent to trying to access `LoadBalancer.backends().get(backendName)` and similar child access scenarios, where children are returned as a `Map`. The Map doesn't throw in the case of a bad name.